### ISSUE TITLE
Feature Request : simplify use of variables and profiles

### ISSUE THREAD

**Issue by lucas-dclrcq** [oknozor/toml-bombadil#21]
In the current state of bombadil the use of variables and profiles is overly complicated :
- we should not need a specific type of global variables (meta)
- profiles are linked to a dot and create a specific command every time, it can rapidly get cluttered

_Proposal_ : 

- Variables can reference any other variable name using a special format
- Meta variables are not necessary anymore so remove them
- Use a "default profile for all dots, hooks, var files etc
- Add the possibility to link dots, hooks and var files to a custom profile (ex: corporate)
- When "activating" the profile it should override the variables of the default profile if existing
- Add a parameter to the link command to enable the user to activate a specific profil (if not specified the default profile is used)
- The profile parameter could take an array to enable the user to combine profiles. (values would be overriden following the profiles order in the array)

_Example :_

bombadil.toml
```
# Default dot for maven
[[dot]]
name = "maven"
source = "maven/settings.xml
target = ".m2/settings.xml

# Dot for corporate profile
[[dot.corporate]]
name = "maven"
source = "maven/settings.corporate.xml" # activating the corporate profile will change the source for settings.xml but not the target

# Variables activated by default
[[var]] 
path = "vars.toml"

[[var]] 
path = "vars2.toml"

# Variables activated for corporate profile, would override existing variables from default profile
[[var.corporate]]
path = "vars.corporate.toml"

# Hook activated by default, will always run
[[hook]]
command  = "sway reload"

# Hook activated for corporate profile, will run only if corporate profile is activated
[[hook.corporate]]
command = "echo i love $$"
```

**Comment by oknozor** [comment]
The problem with this approach is that representing config like this is not straight forward :
```toml
[[dot]]
# ... 

# Dot for corporate profile
[[dot.corporate]]
# ...
```

`[[dot]]` represent an entry in a `Vec<Dot>` and `[[dot.corporate]]` would be deserialized as `HasMap<String, Dot>` we could use the [toml crate](https://crates.io/crates/toml) to handle deserialization manually.

An alternative representation would be to add dots, hooks and vars entries under a nested `Profile` struct this would look like this : 

```toml
# This is the default profile
[dots]
maven = { source = "maven/settings.xml", target = ".m2/settings.xml" }
alacritty = { source = "alacritty/settings.xml", target = ".config/alacritty" }
vars = [ "colors.toml", "env.toml"]
hooks = [ "sway reload", "echo 42"]

# This how we would add profile, no need to repeat inner values that are not overwritten (such as dot.source)
[dots.profiles.corporate]
maven = { source = "maven/settings.corporate.xml" }
vars = [ "colors.toml" ]
hooks = [ "echo \"evil corp \"" ]
```
Intermediate json representation would be this : 

```json
{
  "dots": {
    "hooks": [
      "sway reload",
      "echo 42"
    ],
    "vars": [
      "colors.toml",
      "env.toml"
    ],
    "alacritty": {
      "source": "alacritty/settings.xml",
      "target": ".config/alacritty"
    },
    "maven": {
      "source": "maven/settings.xml",
      "target": ".m2/settings.xml"
    },
    "profiles": {
      "corporate": {
        "hooks": [
          "echo \"evil corp \""
        ],
        "vars": [
          "colors.toml"
        ],
        "maven": {
          "source": "maven/settings.corporate.xml"
        }
      }
    }
  }
} 
```

This is be less verbose but the trade-off would be that all dot entry needs to be named. Overriding profile value would be easier to implement since we just need to match [dots.name] against the default profile.

**Comment by oknozor** [comment]
closed by #29 and #28
